### PR TITLE
XAJ zero params and initial storage_m bug fixes

### PR DESCRIPTION
**Pull request by rlmcdaniel**
[Short description explaining the high-level reason for the pull request]

## Additions

- logic block in Xinanjiang routine to work when parameters are zero.  Assumes 95% runoff in these cases.
- description of gw_storage and soil_storage in config file cat_87_bmi_config_cfe_pass

## Changes

- fixed initial groundwater storage_m and soil water storage_m from [m/m] to [m] by multiplying the decimal fraction provided in the config file by the max storage in their respective reservoirs.

## Testing

1. Passes mass balance checks when the parameters (smcmax and/or soil_storage) are set to zero (catchment 87). Expected change can be seen in runoff
2. Passes mass balance checks with original parameter set (catchment 87)

## Screenshots
Original parameter set
image

smcmax set to zero
image

## Notes

-

## Todos

-

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows project standards (link if applicable)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Target Environment support

- [ ] Windows
- [ ] Linux
- [ ] Browser

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: configs/cat_87_bmi_config_cfe_pass.txt [hunk 1]
```diff
@@ -11,9 +11,15 @@ soil_params.expon_secondary=1.0[]
 max_gw_storage=0.25[m]
 Cgw=1.8e-05[m h-1]
 expon=6.0[]
+
+# gw_storage is the ground water as a decimal fraction of the maximum groundwater storage (max_gw_storage) for the initial timestep
```
Comment (SnowHydrology) [inline configs/cat_87_bmi_config_cfe_pass.txt:15]: These comments are useful, but I think we'd benefit from consistency. I.e., we should comment all params in all configs, or we should save the comments/descriptions for a readme file.

File: src/bmi_cfe.c [hunk 2]
```diff
@@ -586,7 +586,7 @@ int read_init_config_cfe(const char* config_file, cfe_state_struct* model, doubl
             continue;
         }
         if (strcmp(param_key, "gw_storage") == 0) {
-            model->gw_reservoir.storage_m = strtod(param_value, NULL);
+            model->gw_reservoir.storage_m = strtod(param_value, NULL) * model->gw_reservoir.storage_max_m; //edited by RLM to fix units from [m/m] to [m]
```
Comment (SnowHydrology) [inline src/bmi_cfe.c:589]: Good catch!

File: src/bmi_cfe.c [hunk 3]
```diff
@@ -2665,7 +2665,7 @@ extern void init_soil_reservoir(cfe_state_struct* cfe_ptr, double alpha_fc, doub
 
     // Negative amounts are always ignored and just considered emtpy
     if (storage < 0.0) storage = 0.0;
-    cfe_ptr->soil_reservoir.storage_m = storage;
+    cfe_ptr->soil_reservoir.storage_m = storage * cfe_ptr->NWM_soil_params.smcmax * cfe_ptr->NWM_soil_params.D; //edited by RLM to fix units from [m/m] to [m] in storage_m
```
Comment (SnowHydrology) [inline src/bmi_cfe.c:2668]: Good catch again!

**Comment by lcunha0118** [comment]
Rachel and I went through these changes before, so I am in agreement with them. @SnowHydrology I ran in Ngen and got expected results. 
About the comments in the config file, I like the idea of having them on the file, but maybe we should check with Fred since it is his code.

**Comment by SnowHydrology** [comment]
> Rachel and I went through these changes before, so I am in agreement with them. @SnowHydrology I ran in Ngen and got expected results. 

Same here. One thing I've noticed is that we need to specify carrying over updated configuration files when running in Nextgen or you may get wonky results. Ngen by default pulls an out-of-date CFE.

> About the comments in the config file, I like the idea of having them on the file, but maybe we should check with Fred since it is his code.

The comments are useful, but we need consistency. What makes `gw_storage` and `soil_storage` unique relative to other vars and why only in `cat_87_bmi_config_cfe_pass.txt`?

**Review by SnowHydrology — APPROVED** [review]

**Comment by SnowHydrology** [comment]
@lcunha0118 and @rlmcdaniel I moved the config comments to the [README](https://github.com/NOAA-OWP/cfe#configuration-file) that includes param descriptions. If it is desired to have fully commented config files, please modify all and submit a new PR.

Thanks!

**Pull request merged**